character.h

```diff
@@ -0,0 +1,25 @@
+#ifndef CHARACTER_H
+#define CHARACTER_H
+
+#include<iostream>
+#include<memory>
+
+class Character
+{
+public:
+  Character(std::string name, int health, unsigned int damage);
+
+  std::string GetName() const;
+  int GetHealth() const;
+  unsigned int GetDamage() const;
+  bool IsAlive() const;
+  void GetAttacked(const Character& enemy);
+  void Attack(Character& enemy) const;
+  friend std::ostream& operator<<(std::ostream& os, const Character& character);
+private:
+  const std::string name;
+  int health;
```

Comment:
Amugy a health is lehetne akkor mar unsigned, persze akkor a health<0 check az nem mukodne :-)

Reply:
azt pont ezért nem raktuk bele

Reply:
De lehet am a csokkentes elott is tesztelni, nem csak utana ;-)  De jo igy, csak kb ugyanannyira van letjogosultsaga, hogy a hp unsigned int legyen, mint a dmg, szoval ha az egyiknel hasznaljuk, akkor "Szep" volna a masiknal is.